includes/class-filesystem-direct.php

```diff
@@ -0,0 +1,126 @@
+<?php
+/**
+ * The Class for WordPress Direct Filesystem with optimized read and write routines.
+ *
+ * @package aspire-update
+ */
+
+namespace AspireUpdate;
+
+/**
+ * The Class for WordPress Direct Filesystem with optimized read and write routines.
+ */
+class Filesystem_Direct extends \WP_Filesystem_Direct {
+
+	/**
+	 * Reads entire file into an array with options for limiting the number of lines and direction from the the lines are counted.
+	 *
+	 * @since 2.5.0
+	 *
+	 * @param string $file Path to the file.
+	 * @param int    $number_of_lines The number of lines to read. Default is -1 (read all lines).
+	 * @param bool   $count_bottom_to_up Count the lines from the bottom up. Default is false (count from top to bottom).
+	 *
+	 * @return array|false File contents in an array on success, false on failure.
+	 */
+	public function get_contents_array( $file, $number_of_lines = -1, $count_bottom_to_up = false ) {
+		if ( ! $this->exists( $file ) ) {
+			return false;
+		}
+
+		if ( -1 === $number_of_lines ) {
+			return @file( $file );
+		}
+
+		// phpcs:disable WordPress.WP.AlternativeFunctions.file_system_operations_fopen
+		/**
+		 * Extending WP_Filesystem methods for efficiency.  This is a valid use case.
+		 */
+		$handle = @fopen( $file, 'r' );
+		// phpcs:enable
+		if ( ! $handle ) {
+			return false;
+		}
+
+		$lines      = [];
+		$line_count = 0;
+
+		while ( ( $line = fgets( $handle ) ) !== false ) {
```

Comment:
```suggestion
		// phpcs:ignore Generic.CodeAnalysis.AssignmentInCondition.FoundInWhileCondition -- Intentional use.
		while ( ( $line = fgets( $handle ) ) !== false ) {
```
This can raise an issue for `Generic.CodeAnalysis.AssignmentInCondition.FoundInWhileCondition`.

Adding a comment like `// phpcs:ignore <sniff name> -- <reason>` will avoid this. As this only occurs once in the file, there's no need to put a blanket exclusion for this file in the `phpcs.xml.dist` file.